PythonVisualizations/Heap.py

```diff
@@ -0,0 +1,518 @@
+        cellCenter = self.cellCenter2(0)
+        toPositions = (cellCoords, cellCenter)
+        # self.moveItemsOffCanvas(items, N, sleepTime=0.02) ## alternative method to moveItemsBy
+        self.moveItemsTo(items, toPositions)
+        self.list[0] = n
+        # delete the last element from the list
+        del self.list[-1]
+
+
+    def trickleDown(self, cur=0, size=-1):
+        self.startAnimations()
+        callEnviron = self.createCallEnvironment()
+
+        if size == -1: size = len(self.list)
+        top = self.list[cur]  # save the Node at the root
+
+        # while the current Node has at least one child...
+        while cur < size // 2:
+            leftChild = 2 * cur + 1
+            rightChild = leftChild + 1
+            largerChild = leftChild
+            top = self.list[cur]
+            # find smaller child (right child might not exist)
+            if rightChild < size and \
+                    self.list[leftChild] < self.list[rightChild]:
+                largerChild = rightChild
+            # done trickling if top's key is >= the key of larger child
+            if top >= self.list[largerChild]:
+                cur += 1
+
+            else:
+                # shift child up
+                self.swap(cur, largerChild)
+                cur += 1
+
+        self.cleanUp(callEnviron)
+
+
+    def randomFill(self, val):
+        callEnviron = self.createCallEnvironment()
+        self.startAnimations()
+
+        self.size = val
+        del self.list[:]
+        self.display()
+
+        # adding the values to the list
+        for i in range(val):
+            num = random.randint(1, 99)
+
+            # don't move arrow up if the first cell is being filled because it is already pointing there
+            if len(self.list) >= 1:
+                self.moveItemsBy(self.indexDisplay, (0, + self.CELL_HEIGHT))
+
+            cellCoords = self.cellCoords2(len(self.list))  # Color box
+            cellCenter = self.cellCenter2(len(self.list))  # Number in box
+
+            # create new cell and cell value display objects
+            cellCoords = add_vector(cellCoords, (0, 0, 0, self.CELL_BORDER))
+            toPositions = (cellCoords, cellCenter)
+
+            # determine the top left and bottom right positions
+            startPosition = [self.HEAP_X0, 0, self.HEAP_X0, 0]  # Color
+            startPosition = add_vector(startPosition, (0, 0, self.CELL_WIDTH, self.CELL_HEIGHT))  # color`
+            cellPair = self.createCellValue(startPosition, num)
+            callEnviron.add(cellPair)
+            self.moveItemsTo(cellPair, toPositions, steps=self.CELL_HEIGHT, sleepTime=0.01)
+
+            # add a new Drawable with the new value, color, and display objects
+            d = drawable(
+                num, self.canvas.itemconfigure(cellPair[0], 'fill')[-1], *cellPair)
+            self.list.append(d)  # store item at the end of the list
+
+        # finish the animation
+        self.cleanUp(callEnviron)

```

Comment:
Instead of animating the arrival of each of the new item values, let's do something more like the Shuffle animation in SimpleSorting.py.  Have them all arrive at once, maybe in a random cloud, and then slide into position in the array.  Doing them one at a time is too slow for the largest heaps.